src/app/reporting/Engine.h

```diff
@@ -119,6 +121,8 @@ class Engine
      *
      */
     uint32_t mCurReadHandlerIdx = 0;
+
+    ClusterInfo mChangeSetPathList[CHIP_IM_MAX_NUM_SERVER_DIRTY_PATHS];
```

Comment:
I also see this pattern fairly pervasively across the IM when dealing with pooled objects, where the flag for a pooled object being 'in-use' is embedded in the object itself, and done so fairly inconsistently.

In this example here for ClusterInfo, you use the `IsDirty` flag to indicate that that. This is not only a poor name (because it's odd for a dirty path list to _not have a dirty item in it_), but we should really not embed the validity of the object _within the object itself_.

For starters, I'd recommend this move to using the `BitMapObjectPool<T>`, which would permit you to remove the `IsDirty` flag from the ClusterInfo object.

Reply:
just remove this one, and use the existing cluster info pool from interaction model engine, would have seperate pr to update it with object pool